### PR TITLE
added next.js 14 support

### LINKED ISSUE #575: Support NextJS 14
```
npm ERR! code ERESOLVE
--
14:57:32.907 | npm ERR! ERESOLVE could not resolve
14:57:32.908 | npm ERR!
14:57:32.908 | npm ERR! While resolving: next-contentlayer@0.3.4
14:57:32.908 | npm ERR! Found: next@14.0.0
14:57:32.908 | npm ERR! node_modules/next
14:57:32.908 | npm ERR!   next@"^14.0.0" from the root project
14:57:32.908 | npm ERR!
14:57:32.908 | npm ERR! Could not resolve dependency:
14:57:32.908 | npm ERR! peer next@"^12 \|\| ^13" from next-contentlayer@0.3.4
14:57:32.908 | npm ERR! node_modules/next-contentlayer
14:57:32.909 | npm ERR!   next-contentlayer@"^0.3.4" from the root project
14:57:32.909 | npm ERR!
14:57:32.909 | npm ERR! Conflicting peer dependency: next@13.5.6
14:57:32.909 | npm ERR! node_modules/next
14:57:32.909 | npm ERR!   peer next@"^12 \|\| ^13" from next-contentlayer@0.3.4
14:57:32.909 | npm ERR!   node_modules/next-contentlayer
14:57:32.909 | npm ERR!     next-contentlayer@"^0.3.4" from the root project
```

### PR DESCRIPTION
**Pull request by Romstar**
closes #575.

On November 7th I messaged him directly on Discord to become a maintainer for the project or for him to approve of the PR.

I don't think he even read the message.

Maybe we should fork and publish it to NPM?
image

### CODE REVIEW

File: packages/next-contentlayer/package.json
Comment (marcofranssen) [inline packages/next-contentlayer/package.json:67]:
```suggestion
}

```

**Comment by tnnz20** [comment]
Can we installed this versions?

**Comment by ky-zo** [comment]
Please....

**Comment by bodafoo** [comment]
Unprofessional

**Comment by gitgudchannel** [comment]
Please merge it already

**Comment by satish860** [comment]
Is there any issue which needs to be fixed? what is holding up the merge. Happy to help if any.

**Comment by fuma-nama** [comment]
Been using Next.js 14 with Contentlayer for a while, no problem found. Surely this should be merged since package managers like NPM will throw an unmet peer dependencies error due to the incompatible peer dependency version.

**Comment by PieterDePauw** [comment]
@schickling

**Review by Arjun-Sharma1 — APPROVED** [review]

**Comment by blitzhetz** [comment]
If you need to use contentlayer just create next project using npx create-next-app@13 project-name --use-pnpm or without use-pnpm if you want npm. Or switch to 13.5.6 nextjs in package.json and reinstall packages. But anyway this should be merge as soon as possible.

**Comment by anthonyverducci** [comment]
Is contentlayer still actively maintained? I've been excited to add this to my Next.js project the last several weeks, but it seems like the last changes were several months ago in September. Wondering if I need to look into alternate CMS solutions? Happy to help in any way to get this PR merged.

**Comment by michaelangeloio** [comment]
commenting to subscribe for updates

**Comment by Gosev** [comment]
Also commenting to subscribe for updates.

**Review by iammukeshm — APPROVED** [review]

**Comment by Romstar** [comment]
On November 7th I messaged him directly on Discord to become a maintainer for the project or for him to approve of the PR.

image

**Review by andresin87 — APPROVED** [review]

**Comment by andresin87** [comment]
up 🙏

**Comment by Romstar** [comment]
Apparently funding has stopped for the project and Netlify owns it.

The current work around is to add this snippet of code to your root package.json.  I am using turbo repo so it has to be the root package.json and if you aren't using a monorepo then you simply need to add it to your single package.sjon file.

```"overrides": {
    "next-contentlayer": {
      "next": "^14.0.0"
    }
  }```

**Comment by denu5** [comment]
@leerob cant you guys take over :((

**Comment by marcofranssen** [comment]
I understood netlify took over the company @schickling is/was working for. They haven't allocated budget yet to continue the maintenance of contentlayer and therefore it is kind of dead.

**Comment by schickling** [comment]
Hi folks,

Please excuse the late message here and the lack of updates. As others have stated above, the past work on Contentlayer was sponsored by Stackbit which has since been acquired by Netlify who hasn't yet been resuming the sponsorship of the project (still TBD but it's taking a while).

### Why isn't this PR merged yet?

Merging this PR alone most likely won't be enough. There needs to be a new release (with a non-trivial release process) incl. updating transitive dependencies which often cause other parts of Contentlayer to break.

In case there is any other company who'd be willing to sponsor the work on Contentlayer, I'd be [up for a chat](https://twitter.com/schickling). I'd be happy to continue the work on Contentlayer as I'm excited about it and had many plans for it - and it's sad for me personally to see it neglected like this. (As much as I'd like, I can't currently commit to working on Contentlayer in my spare time.)

**Comment by aizk** [comment]
hello anyone can merge???

**Comment by fuma-nama** [comment]
> hello anyone can merge???

Please read the comment above, it won't be merged until someone sponsors this project.

**Comment by ArmantG** [comment]
> closes #575.
> 
> On November 7th I messaged him directly on Discord to become a maintainer for the project or for him to approve of the PR.
> 
> I don't think he even read the message.
> 
> Maybe we should fork and publish it to NPM? image

Fork it! Cant believe this project is standing still like this. I dont have the skills yet to take on a fork of this, but would support anyone who wanted to/could do it!

**Review by gshudhanshu — APPROVED** [review]

**Pull request closed without merging**
